src/Symfony/Component/Validator/Constraints/FileValidator.php

```diff
@@ -229,7 +229,7 @@ private function factorizeSizes(int $size, int|float $limit, bool $binaryFormat)
 
         // Restrict the limit to 2 decimals (without rounding! we
         // need the precise value)
-        while (self::moreDecimalsThan($limitAsString, 2)) {
+        while ($limitAsString < 1 || self::moreDecimalsThan($limitAsString, 2)) {
```

Comment:
instead of this loop comparing a string to an number, a better implementation might be to handle that with a separate loop that the one dealing with decimals:

```php
while ($limit < $coef) {
    $coef /= $coefFactor;
}

$limitAsString = (string) ($limit / $coef);
// Existing logic...
```